contracts/LiquidityManager.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@uniswap/v3-core/contracts/interfaces/IUniswapV3Factory.sol";
+import "@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol";
+import {INonfungiblePositionManager} from "./interfaces/INonfungiblePositionManager.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import {Token} from "./Token.sol";
+
+interface IWETH is IERC20 {
+    function deposit() external payable;
+    function withdraw(uint) external;
+}
+
+interface IERC721Receiver {
+    function onERC721Received(address operator, address from, uint256 tokenId, bytes calldata data) external returns (bytes4);
+}
+
+abstract contract LiquidityManager is IERC721Receiver, Ownable {
+    uint24 public constant UNISWAP_FEE = 3000;
+    int24 private constant MIN_TICK = -887272;
+    int24 private constant MAX_TICK = -MIN_TICK;
+    int24 private constant TICK_SPACING = 60;
+
+    address internal immutable WETH;
+    address public nonfungiblePositionManager;
+    address public uniswapV3Factory;
+
+    constructor(address _uniswapV3Factory, address _nonfungiblePositionManager, address _weth) Ownable(msg.sender) {
+        uniswapV3Factory = _uniswapV3Factory;
+        nonfungiblePositionManager = _nonfungiblePositionManager;
+        WETH = _weth;
+    }
+
+    function onERC721Received(address operator, address from, uint256 tokenId, bytes calldata) external returns (bytes4) {
+        return IERC721Receiver.onERC721Received.selector;
+    }
+
+    function _createLiquilityPool(address tokenAddress) internal returns (address) {
+        IUniswapV3Factory factory = IUniswapV3Factory(uniswapV3Factory);
+
+        address pool = factory.createPool(tokenAddress, WETH, UNISWAP_FEE);
+
+        return pool;
+    }
+
+    function sqrt(uint256 x) internal pure returns (uint256) {
```

Comment:
use sqrt function in UD60X18 from prb-math instead? It is already installed in foundry dependency. Same approximation method, but checks for overflow, uses better initialization value, and limit the number of approximation steps